libraries/BLE/src/BLEScan.cpp

```diff
@@ -107,19 +111,25 @@ void BLEScan::handleGAPEvent(
 					advertisedDevice->setAddress(advertisedAddress);
 					advertisedDevice->setRSSI(param->scan_rst.rssi);
 					advertisedDevice->setAdFlag(param->scan_rst.flag);
-					advertisedDevice->parseAdvertisement((uint8_t*)param->scan_rst.ble_adv, param->scan_rst.adv_data_len + param->scan_rst.scan_rsp_len);
+					if (m_shouldParse) {
+						advertisedDevice->parseAdvertisement((uint8_t*)param->scan_rst.ble_adv, param->scan_rst.adv_data_len + param->scan_rst.scan_rsp_len);
+					} else {
+						advertisedDevice->setPayload((uint8_t*)param->scan_rst.ble_adv, param->scan_rst.adv_data_len + param->scan_rst.scan_rsp_len);
+					}
 					advertisedDevice->setScan(this);
 					advertisedDevice->setAddressType(param->scan_rst.ble_addr_type);
 
-					if (!found) {   // If we have previously seen this device, don't record it again.
+					if (!m_wantDuplicates && !found) {   // If we have previously seen this device, don't record it again.
```

Comment:
What will happen if you want duplicates?
`m_wantDuplicates = true`
In that case this will always return false and device will never be added to vector:
`if (!m_wantDuplicates && !found)`

Reply:
when start continuous scan with callback. vector add allocate without de-allocatted that cause ESP restart.

when not to keep in vector
1. callback was defined.
or 2. wantDuplicates = true
or 3. already in vector (found = true)